CookieListItem is supposed to only expose name and value

annevk opened the pull request:
#### 5bc8fcb69ca17d5ca4f35d609458f1998fc54cb4
<pre>
CookieListItem is supposed to only expose name and value
<a href="https://bugs.webkit.org/show_bug.cgi?id=296878">https://bugs.webkit.org/show_bug.cgi?id=296878</a>

Reviewed by Rupin Mittal.

Remove code we are not planning on shipping. It&apos;s also being removed
from the specification: <a href="https://github.com/whatwg/cookiestore/pull/274">https://github.com/whatwg/cookiestore/pull/274</a>

In order to not reduce test coverage we implement support for
testdriver.js APIs to be able to inspect cookies in web-platform-tests.

Tests are upstreamed and reviewed here: <a href="https://github.com/web-platform-tests/wpt/pull/54129">https://github.com/web-platform-tests/wpt/pull/54129</a>

Canonical link: <a href="https://commits.webkit.org/298384@main">https://commits.webkit.org/298384@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/9eac5d4d82a7c790b145907fc89c614fc8607cfa

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/115378 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/35072 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/25574 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/121468 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/65955 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/fe8f7762-1237-480b-a7ca-13b79449157c) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/117267 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/35731 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/43644 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/87662 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/42357 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/fa48fb93-f164-4877-89ff-01a02dbed11c) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/118326 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/28500 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/103572 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/68056 "Passed tests") | | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/27658 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/21694 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/65125 "Built successfully") | | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/107590 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/97888 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/21807 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/124632 "Built successfully") | | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/113919 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/42317 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/31700 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/96449 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/42685 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/99760 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/96231 "Passed tests") | | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/41461 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/19318 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/38232 "Built successfully") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/18450 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/42200 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/47753 "Built successfully") | [  ~~🛠 jsc-armv7~~](https://ews-build.webkit.org/#/builders/35/builds/139026 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/41702 "Built successfully") | | [  ~~🧪 jsc-armv7-tests~~](https://ews-build.webkit.org/#/builders/35/builds/139026 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/45030 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/43422 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

annevk commented on LayoutTests/http/tests/cookies/cookie-store-get-secure.https.html:
```diff
@@ -1,53 +0,0 @@
-<!DOCTYPE html>
```
It doesn't check that? It only checks secure on CookieListItem.

annevk commented on LayoutTests/platform/mac-wk2/imported/w3c/web-platform-tests/cookiestore/change_eventhandler_for_document_cookie.https.window-expected.txt:
```diff
@@ -1,8 +0,0 @@
-
```
This test is skipped. This file shouldn't have been checked in when I synced last time.

annevk commented on LayoutTests/imported/w3c/web-platform-tests/cookiestore/cookieListItem_attributes.https.any.js:
```diff
@@ -109,15 +82,7 @@ promise_test(async testCase => {
-  for (const key of kCookieListItemKeys) {
-    assert_in_array(key, itemKeys);
-  }
+  assert_cookie_keys(cookie);
```
Yeah, ideally we find some way to test that through WPT, but I'm not sure that's really feasible.

annevk replied:
@RupinMittal I think the bug you filed on `NSHTTPCookie` is wrong. At least per https://httpwg.org/http-extensions/draft-ietf-httpbis-layered-cookies.html once a cookie is stored it always has a host. And I suspect the original tests were also wrong therefore. So I'm not going to add a separate test for this after all and I suggest we WONTFIX that internal bug.

RupinMittal replied:
Ah yes I see it now.

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/dedb74d025e78785a789a49f70ca74f2367e78dc. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/48891

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/111c867dea2ab979a7ff744c5ac1bb6dbf82b0af. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/48891

RupinMittal commented on Source/WebCore/testing/Internals.idl:
```diff
@@ -1311,6 +1326,7 @@ enum ContentsFormat {
 
     undefined setCookie(CookieData cookieData);
     sequence<CookieData> getCookies();
+    sequence<WebDriverCookieData> webDriverGetCookies(Document document);
```
Is there any reason we can't just use `CookieData` and the above functions? Seems like duplicate logic.

RupinMittal commented on Source/WebCore/testing/Internals.h:
```diff
@@ -1349,6 +1349,57 @@ class Internals final
+        bool secure;
+        bool httpOnly;
+        std::optional<double> expiry;
+        String sameSite;
```
`String sameSite { "None"_s };`

RupinMittal reviewed (commented):
Some comments.

annevk commented on Source/WebCore/testing/Internals.h:
```diff
@@ -1349,6 +1349,57 @@ class Internals final
+        { }
+
+        WebDriverCookieData()
+        { }
```
This is used by the bindings code. Not entirely sure in what way, but it doesn't compile without this. `CookieData` has the same.

RupinMittal reviewed (approved):
LGTM. Don't forget to create and add a radar :)

annevk committed:
CookieListItem is supposed to only expose name and value
https://bugs.webkit.org/show_bug.cgi?id=296878

Reviewed by Rupin Mittal.

Remove code we are not planning on shipping. It's also being removed
from the specification: https://github.com/whatwg/cookiestore/pull/274

In order to not reduce test coverage we implement support for
testdriver.js APIs to be able to inspect cookies in web-platform-tests.

Tests are upstreamed and reviewed here: https://github.com/web-platform-tests/wpt/pull/54129

Canonical link: https://commits.webkit.org/298384@main
